js/src/table.ts

```diff
@@ -242,7 +243,7 @@ class FilteredDataFrame implements DataFrame {
         }
         return sum;
     }
-    public *[Symbol.iterator](): IterableIterator<Struct['TValue']> {
+    public *[Symbol.iterator](): IterableIterator<StructValue<T>> {
```

Comment:
@TheNeuralBit since the `Struct` type now takes a generic parameter of `StructData`, I think this can be `Struct<T>['TValue']` right?

Reply:
Ah yeah good point! That will be a more correct typing anyway since it'll include the Row Proxy and the View https://github.com/apache/arrow/pull/2256/files/e82a6935d7c18b477c0ee74295e060e01c91ee65#diff-9b37287b5d48e60adf673234c17c04c5R363